frontend/src/app/word-models/related-words/related-words.component.ts

```diff
@@ -28,15 +32,37 @@ export class RelatedWordsComponent implements OnChanges {
 
     faCheck = faCheck;
 
-    constructor(private wordModelsService: WordmodelsService) { }
+    constructor(
+        route: ActivatedRoute,
+        router: Router,
+        paramService: ParamService,
+        private wordModelsService: WordmodelsService
+    ) {
+        super(route, router, paramService);
+    }
 
     ngOnChanges(changes: SimpleChanges) {
         if (changes.corpus || changes.queryText) {
             this.getData();
         }
     }
 
+    initialize() {
+        
+    }
+
+    teardown() {
+        this.setParams({ neighbours: null });
+    }
+
+    setStateFromParams(params: Params) {
+        if (params.has('neighbours')) {
+            this.neighbours = params.get('neighbours');
+        }
```

Comment:
This would preserve non-default values for `neighbours` if they are removed from the route. If a parameter is included in the route, it should be set from the route in all cases.

```suggestion
        this.neighbours = _.get(params, 'neighbours', 5);
```